From 3a76450c2c430f5ff6a04ac51beea0a6e27f26df Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 14 Jun 2017 17:40:17 +0800
Subject: [PATCH 1890/5242] MLK-15043-2: ASoC: imx-cs42888: fix noise issue
 with FE-BE case

commit  b21ceed8d4f4054c2e110839069041c1e195ddb6 from
https://source.codeaurora.org/external/imx/linux-imx.git

The case is "aplay -Dhw:0,1 -d 5 -r 8000 -f S16_LE -c 9 audio8k16b9c.wav",
which is to playback 9 channel bitstream. But the maximum supported channel
of codec is 8, ALSA didn't return error for this case, but continue to
playback.

The reason is that in FE-BE case, ASLA only get the FE's hw parameter for
constraint list, omit the BE's parameter. This patch is to merge BE's
parameter to FE. in this situation with the 9 channel case, ASLA will
return error "aplay: set_params:1303: Channels count non available"

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit cfe36e2e7fce578eeac9a36eac04db2584e6e92e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-cs42888.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/fsl/imx-cs42888.c b/sound/soc/fsl/imx-cs42888.c
index 43a1584..fc5f109 100644
--- a/sound/soc/fsl/imx-cs42888.c
+++ b/sound/soc/fsl/imx-cs42888.c
@@ -277,6 +277,7 @@ static int be_hw_params_fixup(struct snd_soc_pcm_runtime *rtd,
 		.ignore_pmdown_time = 1,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
+		.dpcm_merged_chan = 1,
 	},
 	{
 		.name = "HiFi-ASRC-BE",
-- 
1.7.9.5

